### PR TITLE
Chore/add phpstan

### PR DESCRIPTION
**Pull request by f-dumas**
* [x] Add phpstan to level "max"in oder to improve the code quality
* [x] Add the checker in the CI
* [x] Fix Unit tests
* [x] Add .PHONY instruction in Makefile

### CODE REVIEW

File: Listener/ConsoleEventsSubscriber.php [hunk 1]
```diff
@@ -44,7 +44,10 @@ public function onCommand(ConsoleEvent $event): void
     {
         $this->startTime = microtime(true);
         $this->eventDispatcher->dispatch(
-            ConsoleCommandMonitoringEvent::createFromConsoleEvent($event, $this->startTime)
+            //ConsoleCommandMonitoringEvent::createFromConsoleEvent($event, $this->startTime)
```
Comment (Mourzouk) [inline Listener/ConsoleEventsSubscriber.php]: Maybe remove this comment ?

File: Metric/Metric.php [hunk 2]
```diff
@@ -96,7 +97,7 @@ public function getResolvedName(): string
     public function getResolvedValue(): string
     {
         if ($this->type === self::METRIC_TYPE_INCREMENT) {
-            return 1;
+            return (string) 1;
```
Comment (embaya) [inline Metric/Metric.php]:
```suggestion
            return "1";
```

File: phpstan.neon.dist [hunk 3]
```diff
@@ -0,0 +1,18 @@
+includes:
+    - 'vendor/phpstan/phpstan-phpunit/extension.neon'
+    - 'vendor/phpstan/phpstan-phpunit/rules.neon'
+
+parameters:
+    checkGenericClassInNonGenericObjectType: false
+    checkMissingIterableValueType: false
+    ignoreErrors:
+        - '#Parameter \#1 \$function of function call_user_func expects callable#'
```
Comment (Brah0um) [inline phpstan.neon.dist:9]: Call the `is_callable` function [here](https://github.com/M6Web/StatsdTagsPrometheusBundle/blob/b0d184c02f1dcf3e410cf328ae068e1699db2a24/Metric/Metric.php#L109) will fix this error

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -30,6 +30,7 @@ private function addMetricsSection(ArrayNodeDefinition $rootNode): void
                     ->children()
                         ->scalarNode('prefix')->end()
                     ->end()
+                ->end()
```
Comment (Oliboy50) [inline DependencyInjection/Configuration.php:33]: 🤯

Comment (f-dumas) [reply DependencyInjection/Configuration.php:33]: :thinking:

File: Metric/Metric.php [hunk 5]
```diff
@@ -112,7 +112,12 @@ public function getResolvedValue(): string
             throw new MetricException(\sprintf('The event class "%s" must have a "%s" method or parameters in order to measure value.', \get_class($this->event), $this->paramValue));
         }
 
-        return $this->correctValue(\call_user_func([$this->event, $this->paramValue]));
+        $callable = [$this->event, $this->paramValue];
+        if (!is_callable($callable)) {
+            throw new \Exception('lol');
+        }
+
+        return $this->correctValue(\call_user_func($callable));
```
Comment (Brah0um) [inline Metric/Metric.php:120]:
```suggestion
        return $this->correctValue($this->event->{$this->paramValue}());
```

**Pull request merged**